internal/reader/rmn_home.go

```diff
@@ -0,0 +1,311 @@
+}
+
+type RmnHomePoller struct {
+	wg                   sync.WaitGroup
+	stopCh               services.StopChan
+	sync                 services.StateMachine
+	contractReader       contractreader.ContractReaderFacade
+	rmnHomeBoundContract types.BoundContract
+	lggr                 logger.Logger
+	mutex                *sync.RWMutex
+	rmnHomeState         rmnHomeState
+	failedPolls          uint
+	pollingDuration      time.Duration // How frequently the poller fetches the chain configs
+}
+
+func NewRMNHomePoller(
+	contractReader contractreader.ContractReaderFacade,
+	rmnHomeBoundContract types.BoundContract,
+	lggr logger.Logger,
+	pollingInterval time.Duration,
+) RMNHome {
+	return &RmnHomePoller{
+		stopCh:               make(chan struct{}),
+		contractReader:       contractReader,
+		rmnHomeBoundContract: rmnHomeBoundContract,
+		rmnHomeState:         rmnHomeState{},
+		mutex:                &sync.RWMutex{},
+		failedPolls:          0,
+		lggr:                 lggr,
+		pollingDuration:      pollingInterval,
+	}
+}
+
+func (r *RmnHomePoller) Start(ctx context.Context) error {
+	r.lggr.Infow("Start Polling RMNHome")
+	return r.sync.StartOnce(r.Name(), func() error {
+		r.wg.Add(1)
+		go r.poll()
+		return nil
+	})
+}
+
+func (r *RmnHomePoller) poll() {
+	defer r.wg.Done()
+	ctx, cancel := r.stopCh.NewCtx()
+	defer cancel()
+	// Initial fetch once poll is called before any ticks
+	if err := r.fetchAndSetRmnHomeConfigs(ctx); err != nil {
+		// Just log, don't return error as we want to keep polling
+		r.lggr.Errorw("Initial fetch of on-chain configs failed", "err", err)
+	}
+
+	ticker := time.NewTicker(r.pollingDuration)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			r.mutex.Lock()
+			r.failedPolls = 0
+			r.mutex.Unlock()
+			return
+		case <-ticker.C:
+			if err := r.fetchAndSetRmnHomeConfigs(ctx); err != nil {
+				r.mutex.Lock()
+				r.failedPolls++
+				r.mutex.Unlock()
+			}
+		}
+	}
+}
+
+func (r *RmnHomePoller) fetchAndSetRmnHomeConfigs(ctx context.Context) error {
+	var versionedConfigWithDigests []rmntypes.VersionedConfigWithDigest
+	err := r.contractReader.GetLatestValue(
+		ctx,
+		r.rmnHomeBoundContract.ReadIdentifier(consts.MethodNameGetAllConfigs),
+		primitives.Unconfirmed,
+		map[string]interface{}{},
+		&versionedConfigWithDigests,
+	)
+	if err != nil {
+		return fmt.Errorf("error fetching RMNHomeConfig: %w", err)
+	}
+
+	if len(versionedConfigWithDigests) != 2 {
+		r.lggr.Warnw("expected 2 RMNHomeConfigs, got", "count", len(versionedConfigWithDigests))
+		return fmt.Errorf("expected 2 RMNHomeConfigs, got %d", len(versionedConfigWithDigests))
+	}
+
+	var primaryConfigDigest, secondaryConfigDigest cciptypes.Bytes32
+
+	// check if the versionesconfigwithdigests are set (can be empty)
+	if versionedConfigWithDigests[0].ConfigDigest == (cciptypes.Bytes32{}) {
+		r.lggr.Warnw("primary config digest is empty")
+	} else {
+		primaryConfigDigest = versionedConfigWithDigests[0].ConfigDigest
+	}
+
```

Comment:
I think we can simplify this with
```
primaryConfigDigest := versionedConfigWithDigests[0].ConfigDigest
if primaryConfigDigest.IsEmpty() {
    log.warn
}
```

Reply:
because it's empty in any case either if we set it or not

Reply:
Valid point